src/integration.test.ts

```diff
@@ -54,4 +55,29 @@ describe("Registry end to end integration tests", () => {
     expect(entry).toBeNull();
     expect(signature).toBeNull();
   });
+
+  it("Should return an error if the timeout is too large", async () => {
+    const { publicKey, privateKey } = genKeyPairAndSeed();
+
+    const entry = {
+      datakey: dataKey,
+      data: "foo",
+      revision: BigInt(0),
+    };
+
+    // Try passing minimum timeout, should have no effect in setEntry.
+    await client.registry.setEntry(privateKey, entry, { timeout: 1 });
+
+    const { entry: returnedEntry } = await client.registry.getEntry(publicKey, dataKey);
+
+    expect(returnedEntry).toEqual(entry);
+
+    // Try getting an entry with an excessive timeout.
+    const { entry: returnedEntry2, signature } = await client.registry.getEntry(publicKey, dataKey, {
+      timeout: MAX_GET_ENTRY_TIMEOUT + 1,
+    });
+
+    expect(returnedEntry2).toBeNull();
+    expect(signature).toBeNull();
```

Comment:
you said this would throw an error ?

Reply:
> Sia will return an error and getEntry will return destructured nulls :)

Currently any error from Sia gets caught and we return nulls, nothing changed there.

Reply:
Ok I will create a follow up for that.